packages/web-components/fast-foundation/src/date-picker/date-picker.ts

```diff
@@ -0,0 +1,1262 @@
+        // Once the textfield is loaded, format its value
+        if (!previous && !!next && this.value) {
+            this.textField.value = this.value;
+        }
+    }
+
+    /**
+     * The hour select listbox
+     * @public
+     */
+    public hourSelect: ListboxElement;
+
+    /**
+     * The minute select listbox
+     * @public
+     */
+    public minuteSelect: ListboxElement;
+
+    /**
+     * The meridian select listbox
+     * @public
+     */
+    public meridianSelect: ListboxElement;
+
+    /**
+     * The Calendar month
+     * @public
+     */
+    @observable
+    public calendarMonth: number = new Date().getMonth() + 1;
+
+    /**
+     * Handles changing the calendar view month
+     * @param previous - the current month value
+     * @param next - the next month value
+     * @public
+     */
+    public calendarMonthChanged(previous: number, next: number) {
+        this.setCalendarTitle();
+    }
+
+    /**
+     * The Calendar year
+     * @public
+     */
+    @observable
+    public calendarYear: number = new Date().getFullYear();
+
+    /**
+     * Handles changing the calendar view year
+     * @param previous - the current calendar year view
+     * @param next - the next calendar year view
+     * @public
+     */
+    public calendarYearChanged(previous: number, next: number) {
+        this.setCalendarTitle();
+    }
+
+    /**
+     * Sets the calendar title text
+     * @public
+     */
+    public setCalendarTitle() {
+        if (this.calendarMonth && this.calendarYear) {
+            this.calendarTitle = this.dateFormatter.getDate(
+                `${this.calendarMonth}/1/${this.calendarYear}`,
+                { month: "long", year: "numeric" }
+            );
+        }
+    }
+
+    /**
+     * The title text for the calendar
+     */
+    @observable
+    public calendarTitle: string = "";
+
+    /**
+     * Keyboard handler for the calendar title
+     * @param e - Keyboard event fired
+     * @returns should bubble
+     * @public
+     */
+    public handleCalendarTitleKeydown(e: KeyboardEvent): void {
+        const key: string = e.key;
+
+        switch (key) {
+            case keyEnter:
+                e.preventDefault();
+                this.monthPickerDisplay();
+                break;
+            case keyArrowDown:
+            case keyArrowUp:
+                this.handleCalendarChange(1, e);
+                break;
+        }
+    }
+
+    /**
+     * Handler for the calendar change buttons
+     * @param direction - previous or next calendar
+     * @param e - the keyboard event
+     * @returns should bubble
+     * @public
+     */
+    public handleCalendarChange(
+        direction: number = 1,
+        e?: KeyboardEvent | MouseEvent | undefined
+    ): void {
+        if (e && e instanceof KeyboardEvent) {
+            const key: string = e.key;
+
+            switch (key) {
+                case keyEnter:
+                case keySpace:
+                    e.preventDefault();
```

Comment:
since this is template bound the event will have preventDefault() called on it automatically.

Reply:
You likely want to return true in cases where you don't handle the key.

Reply:
Fixed.